### PR TITLE
UX: Various improvements to UI

### PR DESCRIPTION
**Pull request by davidtaylorhq**

### CODE REVIEW

File: javascripts/discourse/components/kanban/list.gjs [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    }
+
+    const { topic, oldDefinition, oldRefresh } = this.args.dragData;
+    const thisDefinition = this.args.definition;
+
+    let doUpdate = () => {};
+    let requireConfirmation = settings.require_confirmation;
+    let confirmationMessage = "";
+
+    if (this.kanbanManager.mode === "tags") {
```
Comment (Arkshine) [inline javascripts/discourse/components/kanban/list.gjs:190]: Small typo.
```suggestion
    if (this.kanbanManager.model === "tags") {
```

File: javascripts/discourse/components/kanban/wrapper.gjs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  @action
+  exitFullscreen() {
+    this.kanbanManager.fullscreen = false;
+  }
+
+  @action
+  openSettings(menu) {
+    this.modal.show(KanbanOptionsModal);
+    menu.close();
```
Comment (Arkshine) [inline javascripts/discourse/components/kanban/wrapper.gjs:90]: There is no `menu` in this wrapper context.

File: javascripts/discourse/components/kanban/nav.gjs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import i18n from "discourse-common/helpers/i18n";
+import concatClass from "discourse/helpers/concat-class";
+
+export default class KanbanNav extends Component {
+  <template>
+    <li>
```
Comment (Arkshine) [inline javascripts/discourse/components/kanban/nav.gjs:8]: The button should be rendered based on the `display_categories` setting.

This is a change forgotten from the old code which was using a connector:
```js
shouldRender(args) {
  return displayConnector(args.category?.get("slug"));
},
```

File: javascripts/discourse/initializers/initialize-discourse-kanban.js
Comment (Arkshine) [inline javascripts/discourse/initializers/initialize-discourse-kanban.js:1]: > _Calling transitionTo on a route is deprecated [...]_

```js
return this.transitionTo(
```
->
```js
router: service();
...
this.router.transitionTo(
```

File: common/common.scss
Comment (Arkshine) [inline common/common.scss:1]: The height on `/latest` seems to be a little glitchy (chrome/windows). The category view is fine.
I'm afraid I don't have a suggestion right now.

image

File: common/common.scss [hunk 4]
```diff
@@ -180,3 +239,79 @@
...
+  .list-container {
+    flex-shrink: 1;
+    flex-grow: 1;
+    min-height: 0; // Override auto min-height from flexbox
+  }
+
+  .list-container .row:nth-child(2),
+  .full-width {
+    height: 100%;
+  }
```
Comment (janzenisaac) [inline common/common.scss:266]: This seems like a problem

File: javascripts/discourse/services/kanban-manager.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  get discoveryTag() {
+    return this.discoveryRouteAttribute("tag");
+  }
+
+  get active() {
+    return !!this.currentDescriptor;
+  }
+
+  get currentDescriptor() {
+    return this.discoveryParams && get(this.discoveryParams, "board");
```
[the remark this answers is not in the document]
Comment (janzenisaac) [reply javascripts/discourse/services/kanban-manager.js:66]: This has been resolved in https://github.com/discourse/discourse-kanban-theme/pull/36/commits/d4eb78d9d79ea4ceeaee1b0ea7d916de4116605a

File: javascripts/discourse/components/kanban/wrapper.gjs [hunk 6]
```diff
@@ -11,6 +11,35 @@ import KanbanOptionsModal from "./modal/options";
 import { getOwner } from "@ember/application";
 
 export default class Kanban extends Component {
+  @service kanbanManager;
+  @service modal;
+
+  @tracked dragData;
+
+  get isLegacyTopicList() {
```
Comment (davidtaylorhq) [inline javascripts/discourse/components/kanban/wrapper.gjs:19]: I think we can drop this now. The core topic list refactor was merged a while back, so discourse-compatibility should be all we need.

Should be able to remove the `.kanban-legacy-topic-list` css as well

File: javascripts/discourse/components/kanban/list.gjs [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+  @action
+  addResizeListener(element) {
+    window.addEventListener("resize", () => this.calcHeight(element));
+    this.calcHeight(element);
+  }
+
+  @action
+  removeResizeListener() {
+    window.removeEventListener("resize", this.calcHeight);
+  }
```
Comment (davidtaylorhq) [inline javascripts/discourse/components/kanban/list.gjs:307]: `addEventListener` is being passed an anonymous function, but `removeEventListener` is being passed `this.calcHeight`.

The result is that `removeEventListener` will be a no-op, and the listener will stick around.

Best bet is probably an inline modifier. Something similar [here](https://github.com/discourse/discourse-kanban-theme/blob/94492f8705743e0c6138250a33191defdbb30560/javascripts/discourse/components/kanban/list.gjs#L32-L44).

e.g.

```js
const onWindowReize = modifier((element, [callback]) => {
  const wrappedCallback = () => callback(element);
  window.addEventListener("resize", wrappedCallback);
  
  return () => { // Cleanup callback
    window.removeEventListener("resize", wrappedCallback);
  }
}
```

And then in the template

```
<div {{onWindowResize this.calcHeight}}>
```

File: javascripts/discourse/components/kanban/list.gjs [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+    const mainOutlet = element.closest("#main-outlet");
+    const mainOutletHeight = mainOutlet.getBoundingClientRect().height;
+    const mainOutletPadding = 40;
+    const listControlsHeight = mainOutlet
+      .querySelector(".list-controls")
+      .getBoundingClientRect().height;
+    const listTitleHeight = mainOutlet
+      .querySelector(".list-title")
+      .getBoundingClientRect().height;
+    const topicsHeight = document
```
Comment (davidtaylorhq) [inline javascripts/discourse/components/kanban/list.gjs:286]: Unused variable? 

(I guess that means eslint isn't configured properly for gjs in this repo?)

File: .discourse-compatibility [hunk 9]
```diff
@@ -1,3 +1,4 @@
+< 3.1.0.beta2-dev: c78a3998cdafabde7b9fa2e99ddab8ad2449c994
```
Comment (davidtaylorhq) [inline .discourse-compatibility:1]:
```suggestion
< 3.1.0.beta4-dev: c78a3998cdafabde7b9fa2e99ddab8ad2449c994
```

File: javascripts/discourse/components/kanban/card.gjs [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+import icon from "discourse-common/helpers/d-icon";
+
+export default class KanbanCard extends Component {
+  @service kanbanManager;
+
+  @tracked dragging;
+
+  // TODO - FIX THIS ONCE CORE EXPORTS IT PROPERLY
+  formatDate = getOwner(this).resolveRegistration("helper:format-date");
+  categoryBadge = getOwner(this).resolveRegistration("helper:category-badge");
```
Comment (davidtaylorhq) [inline javascripts/discourse/components/kanban/card.gjs:23]: Core exports these properly now, so we should be able to import as normal helpers in gjs (and update the template to remove `this.` from these two)

File: settings.yml [hunk 11]
```diff
@@ -25,7 +25,9 @@ num_columns:
   default: 3
```
Comment (davidtaylorhq) [inline settings.yml:25]: Let's remove the num_columns settings. Following this change, columns are now fixed-width so this setting is unused.

**Review by davidtaylorhq — COMMENTED** [review]
✅

**Review by janzenisaac — APPROVED** [review]

**Pull request merged**